Lib/ast.py

```diff
@@ -1053,15 +1054,82 @@ def visit_AsyncWith(self, node):
         with self.block(extra=self.get_type_comment(node)):
             self.traverse(node.body)
 
+    def _str_literal_helper(
+        self, string, quote_types=("'", '"', '"""', "'''"), escape=""
+    ):
```

Comment:
There are too many different constants hanging around (all quote types, only multiple quotes etc.), let's make them private globals;
```diff
+_SINGLE_QUOTES = ("'", '"')
+_MULTI_QUOTES = ('"""', "'''")
+_ALL_QUOTES = (*_SINGLE_QUOTES, *_MULTI_QUOTES)
+
 class _Unparser(NodeVisitor):
```

Reply:
I spent some time on it about how we could possibly simplify this interface. From what I see `escape` can be either `"\n\t"` or an empty string, which allows us to reshape the signature like the following;
```diff
     def _str_literal_helper(
-        self, string, quote_types=("'", '"', '"""', "'''"), escape=""
+        self,
+        string,
+        *,
+        escape_special=False,
+        quote_types=_ALL_QUOTES,
     ):
```
The `escape_special` should control the behavior of escape of `\n\t`.